crates/fibre/src/lib.rs

```diff
@@ -151,157 +241,28 @@ impl<Resume, Yield, Return> Suspend<Resume, Yield, Return> {
     /// # Panics
     ///
     /// Panics if the current thread is not executing a fiber from this library.
-    pub fn suspend(&self, value: Yield) -> Resume {
-        self.inner
-            .switch::<Resume, Yield, Return>(RunResult::Yield(value))
+    pub fn suspend(&self, tag: TagId) {
+        let reason = SwitchReason::suspend(tag);
+        self.inner.switch(reason);
     }
 
-    fn execute(
-        inner: imp::Suspend,
-        initial: Resume,
-        func: impl FnOnce(Resume, &Suspend<Resume, Yield, Return>) -> Return,
-    ) {
+    fn execute(inner: imp::Suspend, func: impl FnOnce((), &Suspend) -> ()) {
         let suspend = Suspend {
             inner,
             _phantom: PhantomData,
         };
-        let result = panic::catch_unwind(AssertUnwindSafe(|| (func)(initial, &suspend)));
-        suspend.inner.switch::<Resume, Yield, Return>(match result {
-            Ok(result) => RunResult::Returned(result),
-            Err(panic) => RunResult::Panicked(panic),
-        });
+        // Note that the original wasmtime-fiber crate runs `func` wrapped in
+        // `panic::catch_unwind`, to stop panics from being propagated onward,
+        // instead just reporting parent. We eschew this, doing nothing special
+        // about panics.
+        (func)((), &suspend);
```

Comment:
Yes, that's basically what we discussed [here](https://github.com/wasmfx/wasmfxtime/pull/58#discussion_r1402111803): My plan is to change the `fibre` infrastructure in a follow-up PR so that `func` is just a function pointer, rather than a `FnOnce` closure, and remove the unnecessary `Suspend` argument at that point.

Reply:
Good, thanks!